presto-main/src/main/java/com/facebook/presto/memory/ClusterMemoryManager.java

```diff
@@ -143,21 +143,17 @@ public synchronized void process(Iterable<QueryExecution> queries)
             DataSize sessionMaxQueryMemory = getQueryMaxMemory(query.getSession());
             long queryMemoryLimit = Math.min(maxQueryMemory.toBytes(), sessionMaxQueryMemory.toBytes());
             totalBytes += bytes;
-            if (bytes > queryMemoryLimit) {
-                if (resourceOvercommit(query.getSession())) {
-                    // If a query has requested resource overcommit, only kill it if the cluster has run out of memory
-                    if (outOfMemory) {
-                        DataSize memory = succinctDataSize(bytes, BYTE);
-                        query.fail(new PrestoException(CLUSTER_OUT_OF_MEMORY,
-                                format("The cluster is out of memory, you set %s=true, and your query is using %s of memory, so it was killed.", RESOURCE_OVERCOMMIT, memory)));
-                        queryKilled = true;
-                    }
-                }
-                else {
-                    DataSize maxMemory = succinctDataSize(queryMemoryLimit, BYTE);
-                    query.fail(exceededGlobalLimit(maxMemory));
-                    queryKilled = true;
-                }
+            if (resourceOvercommit(query.getSession()) && outOfMemory) {
```

Comment:
Yes, the original code guaranteed that you got your fair share, but it is buggy and implementing that check correctly would be complicated